streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -301,6 +310,18 @@ private void runLoop() {
         }
     }
 
+    private void updateStandbyTasks() {
+        ConsumerRecords<byte[], byte[]> records = restoreConsumer.poll(0);
```

Comment:
A bit out of topic: since most of the time the restore consumer's position will be very close to the producer appending LOE, to achieve some batching we could set the restore consumer's max bytes to some larger value (by default it is 1K in ```ConsumerConfig.FETCH_MIN_BYTES_CONFIG```)